rust/connlib/clients/android/src/lib.rs

```diff
@@ -376,6 +376,10 @@ fn connect(
         }
     });
 
+    rustls::crypto::ring::default_provider()
+        .install_default()
+        .expect("Calling `install_default` only once per process always succeeds");
```

Comment:
I am not entirely sure about Android here. For the logger, we have this hack where we don't initialise things twice. This might need some manual testing to ensure creating a new session still works and doesn't panic on this.

Reply:
yeah, this could be called more than once in the same process

Reply:
I think the only reason it doesn't happen in macos it because of how network extension works it's a different process each time.

Reply:
Correct - on Android it's possible for multiple `connect` and `disconnect` to be called within a single instance of `TunnelService` because android keeps the service running in some cases.

On Apple, `connect` and `disconnect` are tied to the lifetime of the process because the process lifetime is much more closely coupled to the networkextension lifetime.